vispy/visuals/graphs/graph.py

```diff
@@ -0,0 +1,216 @@
+    def animate(self, value):
+        self._animate = bool(value)
+
+    def animate_layout(self):
+        if self._layout_iter is None:
+            if self._adjacency_mat is None:
+                raise ValueError("No adjacency matrix set yet. An adjacency "
+                                 "matrix is required to calculate the layout.")
+
+            self._layout_iter = iter(self._layout(self._adjacency_mat,
+                                                  self._directed))
+
+        try:
+            node_vertices, line_vertices, arrows = next(self._layout_iter)
+        except StopIteration:
+            return True
+
+        self._nodes.set_data(pos=node_vertices, **self._node_data)
+        self._edges.set_data(pos=line_vertices, arrows=arrows,
+                             **self._arrow_data)
+
+        return False
+
+    def set_final_layout(self):
+        if self._layout_iter is None:
+            if self._adjacency_mat is None:
+                raise ValueError("No adjacency matrix set yet. An adjacency "
+                                 "matrix is required to calculate the layout.")
+
+            self._layout_iter = iter(self._layout(self._adjacency_mat,
+                                                  self._directed))
+
+        # Calculate the final position of the nodes and lines
+        node_vertices = None
+        line_vertices = None
+        arrows = None
+        for node_vertices, line_vertices, arrows in self._layout_iter:
+            pass
+
+        self._nodes.set_data(pos=node_vertices, **self._node_data)
+        self._edges.set_data(pos=line_vertices, arrows=arrows,
+                             **self._arrow_data)
+
+    def reset_layout(self):
+        self._layout_iter = None
+
+    def set_data(self, adjacency_mat=None, **kwargs):
+        if adjacency_mat is not None:
+            if adjacency_mat.shape[0] != adjacency_mat.shape[1]:
+                raise ValueError("Adjacency matrix should be square.")
+
+            self._adjacency_mat = adjacency_mat
+
+        for k in self._arrow_attributes:
+            if k in kwargs:
+                translated = (self._arrow_kw_trans[k] if k in
+                              self._arrow_kw_trans else k)
+
+                setattr(self._edges, translated, kwargs.pop(k))
+
+        arrow_kwargs = {}
+        for k in self._arrow_kwargs:
+            if k in kwargs:
+                translated = (self._arrow_kw_trans[k] if k in
+                              self._arrow_kw_trans else k)
+
+                arrow_kwargs[translated] = kwargs.pop(k)
+
+        node_kwargs = {}
+        for k in self._node_kwargs:
+            if k in kwargs:
+                translated = (self._node_kw_trans[k] if k in
+                              self._node_kw_trans else k)
+
+                node_kwargs[translated] = kwargs.pop(k)
+
+        if len(kwargs) > 0:
+            raise TypeError(
+                "{}.set_data() got invalid keyword arguments: {}".format(
```

Comment:
Although `.format` is stated as the preferred way to do it, it actually seems more common and simpler just to use `%s`. By convention it's what we use, so go ahead and chenge this.